### DISCUSSION TITLE
Repeater options and Node red data

### DISCUSSION THREAD

**Discussion by svdrummer** [timmbogner/Farm-Data-Relay-System#9] [Ideas]
I have been enjoying the progress (out of interest) of this project.

I had a portable ESP now based project where the sender was portable and battery powered. I had a few dead areas, so i made some repeaters which just re-transmitted the data and added a Code to say it was sent via a repeater node.
Ie Sender transmits to unit01 gateway. The repeater also has the gateway code of 01 and resends to gateway01. As they cant send and receive at the same time, I never had any issues.

I have not played around much with node red. Does anyone have an export of one, 

Cheers

**Comment by timmbogner** [comment]
That's pretty interesting! I've got an idea for how to make the sender seek out the nearest gateway and pair with it, but I haven't implemented it yet.

Here is the basic node-red setup:
`[{"id":"66d36c0f.cedf94","type":"influxdb out","z":"d7346a99.716ef8","influxdb":"905dd357.34717","name":"","measurement":"DataReading","precision":"","retentionPolicy":"","database":"database","precisionV18FluxV20":"ms","retentionPolicyV18Flux":"","org":"the_organization","bucket":"bkt","x":760,"y":240,"wires":[]},{"id":"93e9822a.3ad59","type":"mqtt in","z":"d7346a99.716ef8","name":"","topic":"esp/fdrs","qos":"2","datatype":"auto","broker":"c513f7e9.760658","nl":false,"rap":true,"rh":0,"x":170,"y":220,"wires":[["d377f9e0.faef98"]]},{"id":"d377f9e0.faef98","type":"json","z":"d7346a99.716ef8","name":"","property":"payload","action":"obj","pretty":false,"x":290,"y":220,"wires":[["ca383562.4014e8"]]},{"id":"ca383562.4014e8","type":"split","z":"d7346a99.716ef8","name":"","splt":"\\n","spltType":"str","arraySplt":1,"arraySpltType":"len","stream":false,"addname":"","x":410,"y":220,"wires":[["6eaba8dd.429e38"]]},{"id":"6eaba8dd.429e38","type":"function","z":"d7346a99.716ef8","name":"Fields","func":"msg.payload = [{\n    data: msg.payload.data\n},{\n    id: msg.payload.id,\n    type: msg.payload.type\n}]\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":530,"y":220,"wires":[["296d0f4b.37a46","66d36c0f.cedf94"]]},{"id":"296d0f4b.37a46","type":"debug","z":"d7346a99.716ef8","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":670,"y":200,"wires":[]},{"id":"905dd357.34717","type":"influxdb","hostname":"127.0.0.1","port":"8086","protocol":"http","database":"database","name":"","usetls":false,"tls":"d50d0c9f.31e858","influxdbVersion":"2.0","url":"http://localhost:8086","rejectUnauthorized":true},{"id":"c513f7e9.760658","type":"mqtt-broker","name":"","broker":"localhost","port":"1883","clientid":"","usetls":false,"protocolVersion":"4","keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","birthMsg":{},"closeTopic":"","closeQos":"0","closePayload":"","closeMsg":{},"willTopic":"","willQos":"0","willPayload":"","willMsg":{},"sessionExpiry":""},{"id":"d50d0c9f.31e858","type":"tls-config","name":"","cert":"","key":"","ca":"","certname":"","keyname":"","caname":"","servername":"","verifyservercert":false}]`

**Comment by svdrummer** [comment]
As i was running off battery only for the sender, I just wanted it to wake-up and send, then sleep. Sometimes the main receiver would get both signals from the sender, but it didnt care as it was the same data anyway. My sender is in my pocket and i was moving around. I just sent data every thirty seconds. If i am passed out, someone would know is was eith on the floor or..sleeping(In reality i would stand it up.)this was a dead man function. The reason for the repeater was because, in the dead areas, a liss of message from the sender for 60 seconds would trigger a loss of signal alarm. To avoid the loss if signals, i made the repeater. Basically all it does is get what is recieves and re-send, only with an extra byte to say that this signale came via a repeater. The repeater has the same ....(forgot the word..ID? ) as the main device. I thought there would be an issue with it sending to itself, however, it doesnt. My sender also waits for the ACK message from what it is sending to. This is all very fast and more reliable than blueTooth.
THank you for the Node red code. I will try importing that and see what it is doing.


My application, by the way, is for an injury alert as we work by our selves alot and there are lots of snakes here.

**Reply by timmbogner** [reply]
Yo, that is a fantastic application for a system like this! Node-RED would actually be a good platform for the base station as well. There are two ways for FDRS to talk to Node-RED: serial and MQTT. Serial would allow you to communicate directly to the last gateway via USB (instead of WiFi and MQTT). I only mention it because it sounds like a very critical system, and this might make it more reliable. 
What is your base-station setup like?

**Reply by prairietech** [reply]
Snakes!! Yuk!!

**Comment by svdrummer** [comment]
The base station is a series of addressable LEDs, which show the status of
each devices and an alarm sound maker in software. I got side tracked with
trying Bluetooth devices for low power, but the system was reduced range
and not as good as wifi in esp8266. EspNOW was fast and still very good
batter. 1 18650 battery gives a month at 15 second deep sleep wake-ups. For
manual alarm, pin 16 in the 8266 is pulled low, triggering the deep sleep
wake up. The button pulls pin 16 low via a signal diode as it also pulls
another pin low at the same time. On wakeup, the other pin is checked for a
low in that pin. The diode is used so the internal watchdog low from pin 16
is not going to affect anything. I am writing this in my phone, so it is
not the best description as I don't have access to my PC. To much emphasis
in projects is on power. Deep sleep on ESPs is only 20 micro amps. I
develop in the D1Mini. For those field, I use just the 8266 bare bones,
otherwise I will be powering the USB and other interfaces.
The biggest issue was the case. I fit everything in a small section of gas
pipe(20mm) with two end caps. I am making a PCB soon with everything
mounted on one section so it slides inside the tube and it waterproof.
Showing location would be based the repeater location. I am a paramedic by
trade, so I know the value of early intervention in medical out in the
field. People generally don't get time to use radio or cell phones.

**Reply by timmbogner** [reply]
That's really well thought-out and useful. I've considered pipe sections for waterproof enclosures before; it makes a lot of sense!
Have you had any situations where your system was put into action?
